Add join games

zenkbaries opened the pull request:
Added column to games/index.html.erb for 'join' link.

Added code to games#update to add current user to black_player_id.

Added navbar:/ signin/signout/signup, 'join game' link 

Notes:
there is no column for black_player_in in games model. Need to add 'black_player_id' column to game modal?

zenkbaries committed:
Added navbar
Added sign in/sign out/sign up to navbar
Added OrionChess brand to navbar
Added "Join game" to navbar"

Added "join" link in index(list) of games.

zenkbaries committed:
Added link to trigger games#update
Added codes to save current_user to black_player_id

zenkbaries committed:
Merge branch 'master' of github.com:FirehoseProjectTeamOrion/OrionChess into add-join-games

mscoutermarsh commented on app/views/games/index.html.erb:
```diff
@@ -7,17 +7,21 @@
+      if game.available? do
+        <tr>
+          <td><%= game.name %></td>
+          <td><%= link_to 'Join', games, method: :update %></td>
```
Looking at this I have an idea :bulb: :)

For this button, there is a way to pass query parameters along as well.

See if you can update it to send `game: { black_player_id: current_user.id }` to the controller.

Then you wouldn't need to change the controller at all (could keep the existing code that was there before)

mscoutermarsh commented on app/views/layouts/application.html.erb:
```diff
@@ -15,6 +15,57 @@
+            <li role="separator" class="divider"></li>
+            <li><a href="#">One more separated link</a></li>
+          </ul>
+        </li> -->
```
if we don't need this code, can be removed

mscoutermarsh commented:
If we missed `black_player_id` when creating the original migrations.

Could you add a migration to add it? Also double check on `white_player_id`?

zenkbaries committed:
fix migration

zenkbaries committed:
fix 'join' link_to

mscoutermarsh commented on db/schema.rb:
```diff
@@ -17,11 +17,15 @@
-    t.string   "name"
+    t.integer  "game_id"
+    t.integer  "white_player"
+    t.integer  "black_player"
```
Also, to add a column. You should be creating a migration and using `add_column`. So there should be a migration file as well.

Schema.rb is autogenerated

zenkbaries committed:
rename black and white player column

zenkbaries committed:
Add 'current_game' and game_params private method

Had struggle with 'link_to' in line 19 of games/index.html.erb

Needed help with this part which its the only thing that i need to get fixed

zenkbaries committed:
adjust code to conform to recently migrated column rename of black_user_id

zenkbaries committed:
fixing things rubocop telling me to do

zenkbaries committed:
fix link_to 'Join'

zenkbaries committed:
fine tuning 'game#update' method

zenkbaries commented:
i noticed the conflict is "schema.rb" the only conflict. I know that schema.rb is autogenerated so how do I handle it?  i just choose my part of it and let the schema.rb rebuilt itself.  
let me know if this is not the right move or not.  want to get that branch merged and move on.
